### ISSUE TITLE
stack upgrade -v requires using the full child stack name for depend stack variables

### ISSUE THREAD

**Issue by SpComb** [kontena/kontena#2813]
This is inconsistent with `kontena stack install`, and the `-v foo.peers` parameters are just silently ignored, and have no effect, so it's hard to figure it out:

```
$ kontena stack install -v foo.peers=test1 test/deptest.yaml 
...
$ kontena service show deptest-foo/redis | grep PEERS
    - PEERS=test1
```

```
$ kontena stack upgrade -v foo.peers=test2 deptest test/deptest.yaml 
...
$ kontena service show deptest-foo/redis | grep PEERS
    - PEERS=test1
```

```
$ kontena stack upgrade -v deptest-foo.peers=test2 deptest test/deptest.yaml 
...
$ kontena service show deptest-foo/redis | grep PEERS
    - PEERS=test2
```

**Comment by SpComb** [comment]
Ping @kke this makes the new `kontena stack upgrade -v` in 1.4 very difficult to use, and also breaks `kontena stack upgrade --values-from-stack` (#2795). It would be nice to have this fixed for the 1.4.0 final.

**Comment by SpComb** [comment]
Looks like this actually already got fixed in #2823: the old [`dependency_values_from_options(stackname)`](https://github.com/kontena/kontena/pull/2823/files#diff-a7aa222361ef2e4cf3ad3dbb9eb13ea7L125) used the wrong `stackname`, but the new [`loader.flat_dependencies(variables: values_from_options)`](https://github.com/kontena/kontena/pull/2823/files#diff-a7aa222361ef2e4cf3ad3dbb9eb13ea7R39) does the right thing with the `depends.each do |as_name, data| => k.start_with?(as_name + '.')`

The [spec](https://github.com/kontena/kontena/blob/3891ac5ee1d680213ae895e67123345298ad40b1/cli/spec/kontena/cli/stacks/upgrade_command_spec.rb#L107) for this isn't very good though, it uses `-v dep_2.dep_var=1`, which is already the default value for that variable. Considering that unknown variables are ignored, the spec doesn't actually test anything.

Looks like the `kontena stack install` and `kontena stack validate` use `dependency_values_from_options(dependency['name'])` correctly.
